### DISCUSSION TITLE
is it possible to dynamically handle subdomains in a single codebase ?

### DISCUSSION THREAD

**Discussion by sandys** [vercel/next.js#30176] [Help]
hi
so im trying to create a website-builder. Each user will get a unique subdomain.

so let's say the url generated by vercel is https://my-builder.vercel.app/
then i want to be able to handle user1.my-builder.vercel.app, user2.my-builder.vercel.app, user3.my-builder.vercel.app 

is it possible to do that ? remember im not creating this during deployment. I'm creating this on runtime from the dashboard - like if a new user signs up, they get a new subdomain.

It would be very helpful to have a solution that will work locally (for development on my laptop) as well as production on vercel. Eventually the "my-builder.vercel.app" portion will become "mycustomdomain.com"...but the problem statement remains the same.

P.S. my user pages are a mix of static and dynamic components - however i will do whatever makes the subdomain possible. im not too stuck on static vs ssr here. i am happy to take (and pay) for the performance hit, if any

**Comment by iKonrad** [comment]
You'd probably need a custom server.ts file for it to handle incoming requests and identify the subdomain (and then route to a desired page).

So you cannot go serverless (like vercel), but you'll need a dedicated server / VPS to have your server running (`npm run start`)

**Comment by stigkj** [comment]
Have you looked at https://vercel.com/blog/wildcard-domains? There is example code there.

**Comment by sandys** [comment]
Actually I got a reply.

You need to have a custom domain for wildcards.

It doesn't work on the free domain url that vercel gives .

So *.something.vercel.app doesn't work.
*.mydomain.com works.


This is going to be a little problematic for dev teams since we do a lot of
staging deployments by each developer. It would be really helpful if vercel
does open up wildcards on the free domains.

**Comment by addlistener** [comment]
> hi so im trying to create a website-builder. Each user will get a unique subdomain.
> 
> so let's say the url generated by vercel is https://my-builder.vercel.app/ then i want to be able to handle user1.my-builder.vercel.app, user2.my-builder.vercel.app, user3.my-builder.vercel.app
> 
> is it possible to do that ? remember im not creating this during deployment. I'm creating this on runtime from the dashboard - like if a new user signs up, they get a new subdomain.
> 
> It would be very helpful to have a solution that will work locally (for development on my laptop) as well as production on vercel. Eventually the "my-builder.vercel.app" portion will become "mycustomdomain.com"...but the problem statement remains the same.
> 
> P.S. my user pages are a mix of static and dynamic components - however i will do whatever makes the subdomain possible. im not too stuck on static vs ssr here. i am happy to take (and pay) for the performance hit, if any

@sandys have you made it work? it's not just the subdomain as https://vercel.com/blog/wildcard-domains explains but assigning totally different top level domains dynamically right?
